controller/controller_test.go

```diff
@@ -240,6 +240,31 @@ var _ = Describe("Controller", func() {
 		})
 	})
 
+	Describe("getting existing Contrail virtual interface", func() {
+		var testNetwork *types.VirtualNetwork
+		BeforeEach(func() {
+			testNetwork = CreateMockedNetworkWithSubnet(client.ApiClient, networkName, subnetCIDR,
+				project)
+		})
+		Context("when vif already exists in Contrail", func() {
+			It("returns existing vif", func() {
+				testInterface := CreateMockedInterface(client.ApiClient, testNetwork, tenantName,
+					containerID)
+				
+				iface, err := client.GetExistingInterface(testNetwork, tenantName, containerID)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(iface).ToNot(BeNil())
+				Expect(iface.GetUuid()).To(Equal(testInterface.GetUuid()))
+			})
+		})
+		Context("when vif doesn't exist in Contrail", func() {
+			It("returns error", func() {
+				_, err := client.GetExistingInterface(testNetwork, tenantName, containerID)
+				Expect(err).To(HaveOccurred())
+			})
+		})
```

Comment:
Negative test case "when vif doesn't exist in Contrail it does not get created" could be added here as well